lib/web/apiserver.go

```diff
@@ -3434,6 +3439,21 @@ func (h *Handler) WithAuth(fn ContextHandler) httprouter.Handle {
 	})
 }
 
+// WithAuth2 ensures that a request is authenticated
+// for plain old non-AJAX requests (does not check the Bearer header).
+// It enforces CSRF checks.
+func (h *Handler) WithAuth2(fn ContextHandler) httprouter.Handle {
```

Comment:
`WithAuthAndFormCSRF`? 🙈

Or... we could name it `WithAuthOnly` and it would check only if the request is authenticated. Similarly, we would add `WithFormCSRFProtection` that would only verify the form field. 
Then in your handler you could compose it like this `WithAuthOnly(WithFormCSRFProtection(..))`. Thanks to that, you could get rid of this check `if r.Method != http.MethodGet && r.Method != http.MethodHead` because you would add CSRF protection only to POST request (and GET would use only `WithAuthOnly`). 
But idk if it doesn't complicate things more ;)

Reply:
Really the only reason I composed auth & CSRF together here is that there are different signatures at play here (`ContextHandler`, `httprouter.Handle`, `httplib.HandlerFunc`) and they were not trivially composable at the router level.

Reply:
Ah that's bad. What do you think about the rename I proposed?

Reply:
Renamed it to `WithAuthCookieAndCSRF` to indicate that we check the auth cookie (and not the Bearer header), and enforce CSRF check. Hope that's a bit clearer.